### PR TITLE
refactor: Introduce table th element

### PR DESCRIPTION
**Pull request by pan-kot**
### Description

Introduce an element to be shared for all header cells including selection cell needed for the upcoming change: https://github.com/cloudscape-design/components/pull/1480.

### How has this been tested?

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/table/header-cell/th-element.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import { useMergeRefs } from '../../internal/hooks/use-merge-refs';
+import { TableRole, getTableColHeaderRoleProps } from '../table-role';
+
+interface TableThElementProps {
+  className?: string;
+  style?: React.CSSProperties;
+  sortingStatus?: SortingStatus;
+  sortingDisabled?: boolean;
+  hidden?: boolean;
+  colIndex: number;
```
[the remark this answers is not in the document]
Comment (pan-kot) [reply src/table/header-cell/th-element.tsx:18]: I attempted that once but would prefer to keep that refactoring as a separate PR - there are 107 occurrences of `colIndex` in the table momentarily :)

File: src/table/thead.tsx [hunk 2]
```diff
@@ -96,26 +96,17 @@ const Thead = React.forwardRef(
...
-              )}
-              style={stickyStyles.style}
-              ref={stickyStyles.ref}
-              scope="col"
-              {...getTableColHeaderRoleProps({ tableRole, colIndex: 0 })}
+            <TableThElement
+              className={clsx(headerCellClass, selectionCellClass, hidden && headerCellStyles['header-cell-hidden'])}
+              hidden={hidden}
+              tableRole={tableRole}
+              colIndex={0}
```
[the remark this answers is not in the document]
Comment (pan-kot) [reply src/table/thead.tsx:114]: This one is a col index, not a row index. For the selection cell it is always 0, yes, - but is still to be respected for the ARIA prop.

[the remark this answers is not in the document]
Comment (pan-kot) [reply src/table/thead.tsx:114]: This one is a col index, not a row index. For the selection cell it is always 0, yes, - but is still to be respected for the ARIA prop.

[the remark this answers is not in the document]
Comment (timogasda) [reply src/table/thead.tsx:114]: ah right, my bad.

**Review by timogasda — APPROVED** [review]

**Pull request merged**